woocommerce/woocommerce-gutenberg-products-block
# Convert totals in cart and checkout blocks to explicit display format

This PR converts the total value displays in the Cart and Checkout blocks to explicit formats (e.g currency code appended to the end of the prices). The only change needed was to add a property to `TotalsItem` named `extended` with a default value of `false`, then enabling the extended price on `SubTotal` component by adding the `explicit={ true }` attribute. 

This is directly connected and activated if the result of these checks are true:

https://github.com/Automattic/woocommerce-payments/blob/7d4207cf6d54c306af18220ef4a2153852d7a156/includes/class-wc-payments-explicit-price-formatter.php#L53-L91


Fixes https://github.com/Automattic/woocommerce-payments/issues/2648


#### Accessibility




### Screenshots

<table><tr><td>

Cart Block
---

![image](https://user-images.githubusercontent.com/3295/133646943-34004d5b-6970-4a14-b57c-ef603aebffcf.png)
</td><td>

Checkout Block
---

![image](https://user-images.githubusercontent.com/3295/133647031-73f0625b-bf2c-47e4-bf9f-075ada682c14.png)
</td></tr></table>



### Testing

#### Automated Tests

### Manual Testing

How to test the changes in this Pull Request:

1. Create two pages, one with the cart block, and one with the checkout block
2. With some items in the cart (and having WCPay multi-currency active and having some currencies enabled) visit those pages
3. Check if you see the currency code after the total prices.



### Performance Impact



The currency code data was already provided for the component, I just exported a setting to toggle the explicit price format. There should be no difference in terms of performance. 

### Changelog

> Enable explicit prices on totals to match the effects of multi-currency feature, bundled in WC Payments plugin.
